lib/internal/checkUrl.js

```diff
@@ -24,6 +24,25 @@ function checkUrl(link, baseUrl, options, callback)
 		callHandler.async(callback, link);
 		return;
 	}
+
+	if (cache != null)
```

Comment:
Hmm, github is hiding my comments to this line... here's the new requested change:
```js
if (cache!==undefiend && cache!==null)
```
You were right to use strict. Abstract wouldn't be terrible here, but we can avoid a type check and might as well be consistent.